website_sale_stock_control/models/product.py

```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Sergio Teruel <sergio.teruel@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    inventory_availability = fields.Selection([
+        ('never', 'Allow buy products without control'),
+        ('always', 'Deny buy products without stock'),
+    ], string='Stock Control', default='never',
+    )
+    # To be inherit by other modules
+    website_qty_available = fields.Float(
+        related='qty_available', readonly=True, compute_sudo=True)
```

Comment:
If you put compute_sudo, the quantity will be incorrect in case of multi-company.

Reply:
For avoiding to retranslate the texts (string and help) from upstream, which are very long, add both here.